eth/downloader/downloader.go

```diff
@@ -963,6 +971,20 @@ func (d *Downloader) fetchHeaders(p *peerConnection, from uint64, pivot uint64)
 			go p.peer.RequestHeadersByNumber(from, MaxHeaderFetch, 0, false)
 		}
 	}
+	getNextPivot := func() {
+		pivoting = true
+		request = time.Now()
+
+		ttl = d.requestTTL()
+		timeout.Reset(ttl)
+
+		d.pivotLock.RLock()
+		pivot := d.pivotHeader.Number.Uint64()
+		d.pivotLock.RUnlock()
+
+		p.log.Trace("Fetching next pivot header", "number", pivot+uint64(fsMinFullBlocks))
+		go p.peer.RequestHeadersByNumber(pivot+uint64(fsMinFullBlocks), 2, 2, false) // 3 deep pivot to avoid reorgs
```

Comment:
So this becomes requests for 
- `pivot + 64`, 
- `pivot + 66` ? 
Where `pivot` is the `latest` that the peer has, and not the `latest-64`, which the earlier code used?

Reply:
Well, this part of the code never updated the pivot previously. So here we had `latest-64` until sync finished.

Re pivot+64 and pivot+66, yes, that's correct. The idea was that we need to move the pivot by 64. So if the current one is `pivot`, the next one will be `pivot + 64`. We retrieve `pivot + 66` too to ensure that there's at least 2 blocks imported on top and that we don't get into a reorg.

That said, I also get your objection. Instead of having the pivot at `head - 64` and moving when it becomes `head - 128`, the current code leaves it at `head` and moves it to `head -3` when it becomes old. Fair objection. We could instead of `RequestHeadersByNumber(pivot+uint64(fsMinFullBlocks), 2, 2, false)`, do `RequestHeadersByNumber(pivot+uint64(fsMinFullBlocks), 2, fsMinFullBlocks-8, false)`.

This way we'd be requesting `pivot + 64` and `pivot + 120`, and only jump ahead when there's 120 blocks on top of our pivot. Would that logic work?

Reply:
Fixes by changing to the last suggestion from above.